cmd/open.go

```diff
@@ -0,0 +1,130 @@
+package cmd
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+	"github.com/roots/trellis-cli/trellis"
+)
+
+type OpenCommand struct {
+	UI      cli.Ui
+	Trellis *trellis.Trellis
+}
+
+func (c *OpenCommand) Run(args []string) int {
+	if err := c.Trellis.LoadProject(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	commandArgumentValidator := &CommandArgumentValidator{required: 0, optional: 1}
+	commandArgumentErr := commandArgumentValidator.validate(args)
+	if commandArgumentErr != nil {
+		c.UI.Error(commandArgumentErr.Error())
+		c.UI.Output(c.Help())
+		return 1
+	}
+
+	var openArgs = []string{}
+
+	if len(args) == 0 {
+		siteName, siteNameErr := c.Trellis.FindSiteNameFromEnvironment("development", "")
+		if siteNameErr != nil {
+			c.UI.Error(siteNameErr.Error())
+			return 1
+		}
+
+		url := c.Trellis.SiteFromEnvironmentAndName("development", siteName).MainUrl()
+		openArgs = []string{url}
+	} else {
+		value, exists := c.Trellis.CliConfig.Open[args[0]]
+
+		if !exists {
+			c.UI.Error(fmt.Sprintf("Error: shortcut '%s' does not exist. Check your .trellis/cli.yml config file.", args[0]))
+			c.UI.Error(fmt.Sprintf("Valid shortcuts are: %s", strings.Join(openNames(c.Trellis.CliConfig.Open), ", ")))
+			return 1
+		}
+
+		openArgs = []string{value}
+	}
+
+	open := execCommand("open", openArgs, c.UI)
```

Comment:
Does this work on Linux? Or do we need to use `xdg-open` for that?

Reply:
I think I just naively assumed it exists.. but yeah looks like `xdg-open` is what we want. Added a conditional runtime check

Reply:
Perhaps https://github.com/cli/browser is a better choice?

Example: https://github.com/cli/browser/blob/38f8067bcf5aa9757803ff65d054299f244c0f87/examples/Open/main.go

Reply:
Oh good find, yeah might be better to use that!

Reply:
Can't really use that library unfortunately since it means we can't easily test this command as there's no control over the underlying `cmd`. Running a test literally opens a browser window locally 😂